hilti/runtime/src/fiber.cc

```diff
@@ -652,8 +652,8 @@ void detail::trackStack() {
         return;
 
     if ( fiber->type() == Fiber::Type::IndividualStack || fiber->type() == Fiber::Type::SharedStack ) {
-        if ( auto size = fiber->stackBuffer().activeSize(); size > detail::Fiber::_max_stack_size )
-            detail::Fiber::_max_stack_size = size;
+        detail::Fiber::_max_stack_size =
+            std::max<uint64_t>(fiber->stackBuffer().activeSize(), detail::Fiber::_max_stack_size);
```

Comment:
Note there is a difference between these two: the new version performs a write operation every time, whereas the old version just reads most of the time. I don't know if that makes an actual difference in performance, but this code is executed a lot, so I wouldn't mind staying with the original version.

This applies to the other changes in this commit as well, but is probably not important for those.